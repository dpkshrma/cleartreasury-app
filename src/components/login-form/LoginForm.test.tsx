import * as React from "react";
import { fireEvent, screen } from "@testing-library/react";
import { render } from "../../../test/testUtils";
import {
  Default,
  SignInForm,
  NewPasswordForm,
  ErrorForm,
} from "./LoginForm.stories";

describe("LoginForm", () => {
  it("matches default snapshot", () => {
    const { asFragment } = render(<Default {...Default.args} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches sign in form snapshot", () => {
    const { asFragment } = render(<SignInForm {...SignInForm.args} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches new password form snapshot", () => {
    const { asFragment } = render(
      <NewPasswordForm {...NewPasswordForm.args} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches error form snapshot", () => {
    const { asFragment } = render(<ErrorForm {...ErrorForm.args} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should call onSubmit prop on Sign In button click", () => {
    const onSubmit = jest.fn((e) => e.preventDefault());
    render(<SignInForm {...SignInForm.args} onSubmit={onSubmit} />);
    const submitButton = screen.getByRole("button", { name: /sign in/i });
    fireEvent.click(submitButton);
    expect(onSubmit).toBeCalled();
  });

  it("should call onSubmit prop on Set new password button click", () => {
    const onSubmit = jest.fn((e) => e.preventDefault());
    render(<NewPasswordForm {...NewPasswordForm.args} onSubmit={onSubmit} />);
    const submitButton = screen.getByRole("button", {
      name: /set new password/i,
    });
    fireEvent.click(submitButton);
    expect(onSubmit).toBeCalled();
  });
});
