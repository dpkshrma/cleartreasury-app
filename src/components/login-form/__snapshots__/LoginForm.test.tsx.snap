// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm matches default snapshot 1`] = `
<DocumentFragment>
  <form
    class="flex justify-center flex-col space-y-6"
  >
    <div
      class="w-full relative "
    >
      <label
        class="text-base text-gray-700 inline-block mb-2"
        for="email"
        id="email-label"
      >
        Email address
      </label>
      <input
        class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded"
        id="email"
        name="email"
        placeholder="Enter your email"
        type="email"
        value=""
      />
    </div>
    <div
      class="w-full relative "
    >
      <label
        class="text-base text-gray-700 inline-block mb-2"
        for="password"
        id="password-label"
      >
        Password
      </label>
      <input
        class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded"
        id="password"
        name="password"
        placeholder="Enter your password"
        type="password"
        value=""
      />
    </div>
    <a
      class="text-green-700 text-sm mb-16 cursor-pointer"
      href="/reset-password"
    >
      Forgot your password?
    </a>
    <button
      class="inline-flex items-center justify-center gap-2
    border border-transparent
    ring-theme-button-primary
    text-center leading-snug
    rounded-theme-radius
    focus:outline-none focus:ring-1 focus:ring-offset-1 focus:ring-offset-transparent
    transition-all duration-300
    hover:shadow-md hover:cursor-pointer bg-theme-button-primary text-theme-color-on-primary hover:bg-theme-button-primary-hover py-3 px-8"
    >
      Sign in
    </button>
  </form>
</DocumentFragment>
`;

exports[`LoginForm matches error form snapshot 1`] = `
<DocumentFragment>
  <form
    class="flex justify-center flex-col space-y-6"
  >
    <div
      class="rounded-md p-4 border border-theme-color-critical text-theme-color-critical bg-theme-color-critical-background"
    >
      <div
        class="flex"
      >
        <div
          class="text-theme-color-critical h-6 w-6 flex-shrink-0"
        >
          <svg
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
        <p
          class="text-sm ml-3"
        >
          User does not exists
        </p>
      </div>
    </div>
    <div
      class="w-full relative "
    >
      <label
        class="text-base text-gray-700 inline-block mb-2"
        for="email"
        id="email-label"
      >
        Email address
      </label>
      <input
        class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded"
        id="email"
        name="email"
        placeholder="Enter your email"
        type="email"
        value=""
      />
    </div>
    <div
      class="w-full relative "
    >
      <label
        class="text-base text-gray-700 inline-block mb-2"
        for="password"
        id="password-label"
      >
        Password
      </label>
      <input
        class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded"
        id="password"
        name="password"
        placeholder="Enter your password"
        type="password"
        value=""
      />
    </div>
    <a
      class="text-green-700 text-sm mb-16 cursor-pointer"
      href="/reset-password"
    >
      Forgot your password?
    </a>
    <button
      class="inline-flex items-center justify-center gap-2
    border border-transparent
    ring-theme-button-primary
    text-center leading-snug
    rounded-theme-radius
    focus:outline-none focus:ring-1 focus:ring-offset-1 focus:ring-offset-transparent
    transition-all duration-300
    hover:shadow-md hover:cursor-pointer bg-theme-button-primary text-theme-color-on-primary hover:bg-theme-button-primary-hover py-3 px-8"
    >
      Sign in
    </button>
  </form>
</DocumentFragment>
`;

exports[`LoginForm matches new password form snapshot 1`] = `
<DocumentFragment>
  <form
    class="flex justify-center flex-col space-y-6"
  >
    <div
      class="w-full relative "
    >
      <label
        class="text-base text-gray-700 inline-block mb-2"
        for="email"
        id="email-label"
      >
        Email address
      </label>
      <input
        class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded"
        id="email"
        name="email"
        placeholder="Enter your email"
        type="email"
        value=""
      />
    </div>
    <div
      class="w-full relative "
    >
      <label
        class="text-base text-gray-700 inline-block mb-2"
        for="newPassword"
        id="newPassword-label"
      >
        Change your password
      </label>
      <input
        class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded"
        id="newPassword"
        name="newPassword"
        placeholder="Enter your new password"
        type="password"
        value=""
      />
    </div>
    <button
      class="inline-flex items-center justify-center gap-2
    border border-transparent
    ring-theme-button-primary
    text-center leading-snug
    rounded-theme-radius
    focus:outline-none focus:ring-1 focus:ring-offset-1 focus:ring-offset-transparent
    transition-all duration-300
    hover:shadow-md hover:cursor-pointer bg-theme-button-primary text-theme-color-on-primary hover:bg-theme-button-primary-hover py-3 px-8"
    >
      Set new password
    </button>
  </form>
</DocumentFragment>
`;

exports[`LoginForm matches sign in form snapshot 1`] = `
<DocumentFragment>
  <form
    class="flex justify-center flex-col space-y-6"
  >
    <div
      class="w-full relative "
    >
      <label
        class="text-base text-gray-700 inline-block mb-2"
        for="email"
        id="email-label"
      >
        Email address
      </label>
      <input
        class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded"
        id="email"
        name="email"
        placeholder="Enter your email"
        type="email"
        value=""
      />
    </div>
    <div
      class="w-full relative "
    >
      <label
        class="text-base text-gray-700 inline-block mb-2"
        for="password"
        id="password-label"
      >
        Password
      </label>
      <input
        class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded"
        id="password"
        name="password"
        placeholder="Enter your password"
        type="password"
        value=""
      />
    </div>
    <a
      class="text-green-700 text-sm mb-16 cursor-pointer"
      href="/reset-password"
    >
      Forgot your password?
    </a>
    <button
      class="inline-flex items-center justify-center gap-2
    border border-transparent
    ring-theme-button-primary
    text-center leading-snug
    rounded-theme-radius
    focus:outline-none focus:ring-1 focus:ring-offset-1 focus:ring-offset-transparent
    transition-all duration-300
    hover:shadow-md hover:cursor-pointer bg-theme-button-primary text-theme-color-on-primary hover:bg-theme-button-primary-hover py-3 px-8"
    >
      Sign in
    </button>
  </form>
</DocumentFragment>
`;
